var/spack/repos/builtin/packages/bohrium/package.py

```diff
@@ -0,0 +1,170 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Bohrium(CMakePackage):
+    """Library for automatic acceleration of array operations"""
+
+    homepage    = "http://bh107.org"
+    url         = "https://github.com/bh107/bohrium/archive/v0.8.9.tar.gz"
+    maintainers = ['mfherbst']
+
+    #
+    # Versions
+    #
+    version("develop", git="https://github.com/bh107/bohrium.git",
+            branch="master")
+
+    #
+    # Variants
+    #
+    variant("cuda", default=True,
```

Comment:
please try using 
```
from spack.build_systems.cuda import CudaPackage

class Bohrium(CMakePackage, , CudaPackage):
```
which will have this variant already, together with `cuda_arch` and also relevant constraints for compiler / cuda versions.

You can look at `dealii` to see how it's used.

Reply:
Right now there is no way to pass the cuda flags to Bohrium, so I'll disable this feature for now. (I am also not a cuda expert, so I have to talk to upstream to figure out how to do that properly)

Reply:
ok